client/src/components/HeaderSearch/index.tsx

```diff
@@ -9,29 +9,46 @@ import styles from './styles.module.scss';
 
 const HeaderSearch = () => {
   const { setCoordinates, setStatusMsg } = useContext(LocationContext);
+  const [inputValue, setInputValue] = useState<string>('');
   const navigate = useNavigate();
+
+  const handleChange = event => {
+    if (/^[A-Za-z]*$/.test(event.target.value)) {
+      setInputValue(event.target.value);
+    }
+  };
+
   const handleSubmit = async e => {
     e.preventDefault();
-    const locationData = await getLocationByQuery(e.target[0].value);
-    if (locationData.locations.length === 0) {
-      setStatusMsg('Sorry, your entered location does not exists');
-      setCoordinates(null);
-    } else {
-      const coords = {
-        lat: locationData.locations[0].lat,
-        lon: locationData.locations[0].lon
-      };
-      setCoordinates(coords);
-      setStatusMsg(null);
-      e.target[0].value = '';
-      navigate(`/details/${locationData.locations[0].name}`);
+    if (inputValue) {
```

Comment:
please either show error for empty input or make submit button disabled to not confuse users that clicking button with empty search and nothing happens :)